enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedPreview.tsx

```diff
@@ -33,8 +36,15 @@ export const SdkIframeEmbedPreview = () => {
         script.onload = () => {
           const { MetabaseEmbed } = window["metabase.embed"];
 
+          // If persisted user settings contains non-dashboard ids,
+          // remove the fallback dashboard from the initial state.
+          if (settings.questionId || settings.template) {
+            delete settings.dashboardId;
+          }
```

Comment:
In our initial state, we have to define `dashboardId` to satisfy TypeScript types. If there is already a `questionId` or `template` defined, then we should remove the default fallback.

https://github.com/metabase/metabase/blob/8feca8c5d43d03f96cb04fed0e4bdf37867ef19e/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetupProvider.tsx#L48-L49